frontend/packages/core/src/Charts/linearTimeline.tsx

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import {
+  CartesianGrid,
+  Legend,
+  ResponsiveContainer,
+  Scatter,
+  ScatterChart,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from "recharts";
+
+import { calculateDomainEdges, calculateTicks, localTimeFormatter } from "./helpers";
+
+export interface LinearTimelineDataPoint {
+  timestamp: number;
+  // more to come
+}
+
+// Note that shape can be a set of shapes denoted by preset strings ("circle", "square", etc.)
+// or a custom SVG element
+export interface LinearTimelineDataPoints {
+  points: LinearTimelineDataPoint[];
+  shape?: any;
+  color?: string;
+}
+
+export interface LinearTimelineData {
+  [lane: string]: LinearTimelineDataPoints;
+  // ...
+}
+
+export interface LinearTimelineProps {
+  data: LinearTimelineData;
+  xAxisDataKey: string;
+  regularIntervalTicks?: boolean;
```

Comment:
document this prop, not obvious at first glance what the behavior difference is for true v false.